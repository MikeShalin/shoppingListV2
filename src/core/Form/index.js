/** @flow **/
import * as React from 'react'
import {
  Button,
  Checkbox,
  Form,
  Segment,
} from 'semantic-ui-react'

import Wrapper from 'Core/styled/Wrapper'

type TForm = {
  login: {
    label: string,
    placeholder: string,
    onChange: (value: {}) => void,
    name: string,
  },
  withTerms?: {
    label: string,
    onChange: () => mixed,
    checked: boolean,
  },
  password: {
    label: string,
    placeholder: string,
    onChange: (value: {}) => void,
    name: string,
  },
  onSubmit: () => boolean,
  children: *,
  success: boolean,
  error: boolean,
  loading: boolean,
}

const MainForm = ({
                    login,
                    withTerms,
                    password,
                    onSubmit,
                    children,
                    success,
                    error,
                    loading,
                  }: TForm) => (
  <Wrapper>
    <Segment>
      <Form
        onSubmit={onSubmit}
        success={success}
        error={error}
        loading={loading}
      >
        <Form.Field>
          <Form.Input {...login}/>
        </Form.Field>
        <Form.Field>
          <Form.Input {...password}/>
        </Form.Field>
        <Form.Field>
          {
            withTerms && <Checkbox {...withTerms}/>
          }
        </Form.Field>
        <Button
          color='blue'
          type='submit'
        >Submit</Button>
      </Form>
      {
        (success || error) && children
      }
    </Segment>
  </Wrapper>
)

export default MainForm