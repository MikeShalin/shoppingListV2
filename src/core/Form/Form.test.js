import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import Form from './index'

describe('Form component', () => {
  it('equal early snapshots with terms and conditions', () => {
    const props ={
      login: {
        label: 'First Name',
        placeholder: 'First Name',
      },
      password: {
        label: 'Enter Password',
        placeholder: 'Enter Password',
      },
      withTerms: true,
    }

    const tree = shallow(<Form {...props} />)
    expect(shallowToJson(tree)).toMatchSnapshot()
  })
  it('equal early snapshots without terms and conditions', () => {
    const props ={
      login: {
        label: 'First Name',
        placeholder: 'First Name',
      },
      password: {
        label: 'Enter Password',
        placeholder: 'Enter Password',
      },
    }

    const tree = shallow(<Form {...props} />)
    expect(shallowToJson(tree)).toMatchSnapshot()
  })
})