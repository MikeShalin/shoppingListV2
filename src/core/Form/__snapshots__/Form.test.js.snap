// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component equal early snapshots with terms and conditions 1`] = `
<styled.div>
  <Segment>
    <Form
      as="form"
    >
      <FormField>
        <FormInput
          as={[Function]}
          control={[Function]}
          label="First Name"
          placeholder="First Name"
        />
      </FormField>
      <FormField>
        <FormInput
          as={[Function]}
          control={[Function]}
          label="Enter Password"
          placeholder="Enter Password"
        />
      </FormField>
      <FormField>
        <Checkbox
          type="checkbox"
        />
      </FormField>
      <Button
        as="button"
        color="blue"
        type="submit"
      >
        Submit
      </Button>
    </Form>
  </Segment>
</styled.div>
`;

exports[`Form component equal early snapshots without terms and conditions 1`] = `
<styled.div>
  <Segment>
    <Form
      as="form"
    >
      <FormField>
        <FormInput
          as={[Function]}
          control={[Function]}
          label="First Name"
          placeholder="First Name"
        />
      </FormField>
      <FormField>
        <FormInput
          as={[Function]}
          control={[Function]}
          label="Enter Password"
          placeholder="Enter Password"
        />
      </FormField>
      <FormField />
      <Button
        as="button"
        color="blue"
        type="submit"
      >
        Submit
      </Button>
    </Form>
  </Segment>
</styled.div>
`;
